### PR TITLE
feat(android): add `onControlsVisiblityChange`

### LINKED ISSUE #2686: Controls visible callback
# Feature Request
Callback for when controls show and hide. 
Similar to `onShowControls` and `onHideControls` on `react-native-video-controls`. 

## Why it is needed
I need to add a custom button to the video controls. This callback would allow hiding and showing it. 

I looked at `react-native-video-controls` and `react-native-media-controls` for solving this need, but I also need subtitle support that `react-native-video` has baked in.

### PR DESCRIPTION
**Pull request by ashnfb**
## Summary
A callback event for when the Android Exoplayer controls show and hide.
Issue request: https://github.com/TheWidlarzGroup/react-native-video/issues/2686

### Motivation
Since the Android video player LegacyControls have limited controls (play, pause, rewind, forward, scrubber, etc., and do not provide an interface for changing the Captions or Audio of a stream, or implementing additional buttons), having an event allows a consumer class to overlay buttons / information at the same time as controls are shown.

### Changes
Adds onControlsVisibilityChange to VideoEventEmitter
Adds a visibility listener to the playerControlView in ReactExoplayerview to trigger the new event

Not available for iOS, due to no public methods for monitoring native control visibility

## Test plan

### CODE REVIEW

**Comment by ashnfb** [comment]
@freeboub, any idea how I can fix this ESLint error?  I've tried a few approaches

**Comment by freeboub** [comment]
@ashnfb  did you try : "yarn lint" ?

File: src/Video.tsx [hunk 1]
```diff
@@ -459,6 +454,20 @@ const Video = forwardRef<VideoRef, ReactVideoProps>(
       [onAspectRatio],
     );
 
+    const _onPlaybackRateChange = useCallback(
```
Comment (freeboub) [inline src/Video.tsx]: can you please move back _onPlaybackRateChange at its initial position in the file ? no need for you to be owner of this function :)

**Comment by freeboub** [comment]
I added 1 minor comment, but code looks good ! Thank you

**Comment by ashnfb** [comment]
@KrzysztofMoch @freeboub, corrections made.  I haven't been involved in this project in a while, may I ask what steps are next?

**Comment by KrzysztofMoch** [comment]
If all checks will be green, and @freeboub will approve it (as he requested change) - we can go with it and merge it

**Review by freeboub — APPROVED** [review]

**Pull request merged**